app/utils/id_generator.py

```
@@ -0,0 +1,163 @@
+        try:
+            with lock:
+                if not os.path.exists(self.counters_file):
+                    initial_counters = {
+                        "user_id": 0,
+                        "post_id": 0,
+                        "comment_id": 0
+                    }
+                    with open(self.counters_file, 'w', encoding='utf-8') as f:
+                        json.dump(initial_counters, f, indent=2)
+        except Timeout:
+            raise TimeoutError("ID 카운터 파일 접근 권한을 얻지 못했습니다.")
```

Ruff (0.14.11)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)